packages/opencensus-instrumentation-mongodb/test/test-mongodb.ts

```diff
@@ -0,0 +1,280 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as assert from 'assert';
+import {accessSync} from 'fs';
+import * as http from 'http';
+import * as mocha from 'mocha';
+import * as mongodb from 'mongodb';
+import * as semver from 'semver';
+
+import {plugin} from '../src/';
+import {MongoDBPlugin} from '../src/';
+
+type MongoDBAccess = {
+  client: mongodb.MongoClient,
+  collection: mongodb.Collection
+};
+
+class RootSpanVerifier implements types.SpanEventListener {
```

Comment:
Can you comment on each class and function what they are doing?

Reply:
Sure. Please let me know if they are not clear enough.